### ISSUE TITLE
Deprecation warnings on default props

### ISSUE THREAD

**Issue by ElmarFrerichs** [adazzle/react-data-grid#1403]
### Which version of React JS are you using?

✅ Officially supported ✅
- [ ] v15.4.x

⚠️ Not officially supported, expect warnings ⚠️
- [ ] v15.5.x
- [x] v15.6.x

☣️ Not officially supported, expect warnings and errors ☣️
- [ ] v16.x.x

---

### Which browser are you using?

✅ Officially supported ✅
- [ ] IE 9 / IE 10 / IE 11
- [ ] Edge
- [x] Chrome

⚠️ Not officially supported, but "should work" ⚠️
- [ ] Firefox
- [ ] Safari

---

### I'm submitting a ...

- [x] 🐛 Bug Report
- [ ] 💡 Feature Request

---

### Issue Details

Since at least version 5.0.3 of react-data-grid, I get deprecation warnings for every use of the ReactDataGrid for using the deprecated `enableRowSelect` and `rowScrollTimeout` props, without me ever using them. I think it was caused by #1254.

Call in my code:
```
<ReactDataGrid
  onGridSort={this.handleGridSort}
  columns={this.state.columns}
  rowGetter={this.rowGetter}
  rowsCount={this.getSize()}
  minHeight={currContainerHeight-4}
  onColumnResize={this.onColumnResize}
  headerRowHeight={headerRowHeight} />
```

Two warnings show up in my console:
```
> enableRowSelect has been deprecated and will be removed in a future version. Please use rowSelection instead
> undefined
```

Other deprecated props like `onRowUpdated` don't show a warning, so I suspect it's got something to do with the defaultProps of `enableRowSelect` and `rowScrollTimeout`. The package react-is-deprecated has a [closed issue](https://github.com/aweary/react-is-deprecated/issues/5) on this, the defaultProps are passed through the propTypes as well so the deprecationWarning will trigger.

The warning message for rowScrollTimeout only has `undefined` as text because it's propType definition is missing the `deprecationWarning()` call, but that's another story.

https://github.com/adazzle/react-data-grid/blob/283a83c4b3678eeeb2f6a177e8bda300e44f40a9/packages/react-data-grid/src/ReactDataGrid.js#L132

I didn't digg deep enough into react-data-grid yet to see if just removing the 2 defaultProps will cause any side effects, in my case it worked.

Thanks!

PS: Is React 15.4 still the only officially supported version?

**Comment by shreyamdg** [comment]
Same messages here!

**Comment by ssurve** [comment]
+1

**Comment by espenmeidell** [comment]
+1

**Comment by daneast915** [comment]
+1 in @bentley/ui-components
Using react-data-grid 6.0.1. Please fix in 6.0.2.

**Comment by kristian2x** [comment]
+1 using 5.03

**Comment by piehouserat** [comment]
+1 using latest

**Comment by alexchen1875** [comment]
+1 using 
```
"react-data-grid": "^6.0.2-0",
"react-data-grid-addons": "^6.0.2-0",
```

**Comment by ihomp** [comment]
hay stale bot "react-data-grid": "^6.1.0-0", still has this issue. PR is seems like ready to fix it though.

**Comment by piehouserat** [comment]
looks like they're actively working on a new major release which i'm sure will include this PR -> https://github.com/adazzle/react-data-grid/commits/next

doubt they will include this as part of a minor release before then but I could be wrong

**Comment by rossjp** [comment]
+1 using 6.1.0

**Comment by brianchristensen** [comment]
+1 using
```
"react-data-grid": "^6.1.0",
"react-data-grid-addons": "^6.1.0",
```

**Comment by ryanakallis** [comment]
Awesome that y'all are sorting out console warnings (^^,)

Workaround:
Set prop 'enableRowSelect' to null :
```
<ReactDataGrid
  columns={this.state.columns}
  rowGetter={this.rowGetter}
  rowsCount={this.getSize()}
  _enableRowSelect={null}_
  />
```
The warning 
> undefined 
that ElmarFrerichs mentions with this error is still there, no idea what that is.

**Comment by ryanakallis** [comment]
Awesome that y'all are sorting out console warnings (^^,)

**Workaround**:
Set prop 'enableRowSelect' to null :
```
<ReactDataGrid
  columns={this.state.columns}
  rowGetter={this.rowGetter}
  rowsCount={this.getSize()}
  _enableRowSelect={null}_
  />
```

The warning ElmarFrerichs mentions :
`undefined`
is still there, no idea what that is.

**Comment by m1tttt4** [comment]
@ryanakallis 
> **Workaround**:
> Set prop 'enableRowSelect' to null

Same workaround clears the other undefined caused by rowScrollTimeout as @ElmarFrerichs suspected
 
 ```
<ReactDataGrid
   enableRowSelect={null}
   rowScrollTimeout={null}
/>
```

**Comment by kaminskypavel** [comment]
yup still happening by default at 
```
    "react": "^16.8.6",
    "react-data-grid": "^6.1.0",
    "react-data-grid-addons": "^6.1.0",
```

@m1tttt4  solution will break current types definitions with \

> Error:(127, 11) TS2322: Type 'null' is not assignable to type 'boolean | "single" | "multi" | undefined'.

**Comment by ElmarFrerichs** [comment]
Resolved in version 7.0.0-alpha.18, I'll reopen it if it happens again.
